components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/client/TaskOperationsImpl.java

```diff
@@ -1697,6 +1699,16 @@ private String getCaller() {
                     "performing the task operation!");
         }
 
+        List<String> roleList = pqe.getRoleNameListForUser(pqe.getLoggedInUser());
```

Comment:
What are we actually doing here? We are taking loggedInUser -> rolesAssignedToTheUser -> AllUsersThatAllRolesAreAssignedTo and if atleast one user matches with the loggedInUser we return the matched user, if not, we return the logged in user?

Reply:
Fixed